R/PipeOpTaskSurvClassif.R

```diff
@@ -0,0 +1,133 @@
+#' @title PipeOpTaskSurvClassif
+#' @name mlr_pipeops_trafotask_survclassif
+#' @template param_pipelines
+#'
+#' @description
+#' Transform [TaskSurv] to [TaskClassif][mlr3::TaskClassif] by (describe the method, add one ref?)
+#'
+#' @section Input and Output Channels:
+#' Input and output channels are inherited from [PipeOpTaskTransformer].
+#'
+#' The output is the input [TaskSurv] transformed to a [TaskClassif][mlr3::TaskClassif]
+#' as well as the transformed data during prediction.
+#'
+#' @section State:
+#' The `$state` is a ...
+#'
+#' @section Parameters:
+#' The parameters are
+#'
+#' * `cut :: numeric()`\cr
+#' Split points, used to partition the data into intervals.
+#' If unspecified, all unique event times will be used.
+#' If `cut` is a single integer, it will be interpreted as the number of equidistant
+#' intervals from 0 until the maximum event time.
+#' * `param max_time :: numeric(1)`\cr
+#' If cut is unspecified, this will be the last possible event time.
+#' All event times after max_time will be administratively censored at max_time.
+#' Needs to be greater than the minimum event time.
+#'
+#' @examples
+#' \dontrun{
+#' if (requireNamespace("mlr3pipelines", quietly = TRUE)) {
+#'   library(mlr3)
+#'   library(mlr3pipelines)
+#'
+#'   task = tsk("lung")
+#'   po = po("trafotask_survclassif")
+#'   po$train(list(task))
+#'   po$predict(list(task))[[1]]
+#' }
+#' }
+#' @family PipeOps
+#' @family Transformation PipeOps
+#' @include PipeOpPredTransformer.R
+#' @export
+PipeOpTaskSurvClassif = R6Class(
+  "PipeOpTaskSurvClassif",
+  inherit = PipeOpTaskTransformer,
+
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function(id = "trafotask_survclassif") {
+      param_set = ps(
+        cut = p_uty(default = NULL),
+        max_time = p_dbl(default = NULL, special_vals = list(NULL))
+      )
+      super$initialize(
+        id = id,
+        param_set = param_set,
+        input = data.table::data.table(
+          name    = "input",
+          train   = "TaskSurv",
+          predict = "TaskSurv"
+        ),
+        output = data.table::data.table(
+          name    = c("output", "transformed_data"),
+          train   = c("TaskClassif", "data.frame"),
+          predict = c("TaskClassif", "data.frame")
+        )
+      )
+    }
+  ),
+
+  private = list(
+    .train = function(input) {
+      task = input[[1]]
+      cut = assert_numeric(self$param_set$values$cut, null.ok = TRUE, lower = 0)
+      max_time = self$param_set$values$max_time
+
+      time_var = task$target_names[1]
```

Comment:
consider refactoring this to `task$status()` and `task$times()` or `task$unique_even_times()` maybe you need some of these?

Reply:
rather than getting the variable and calling `task$data()` multiple times